contracts/interchain-token/src/contract.rs

```diff
@@ -1,16 +1,42 @@
 use axelar_soroban_std::ensure;
-use soroban_sdk::{contract, contractimpl, Address, Env};
+use soroban_sdk::token::{self, Interface as _};
+use soroban_sdk::{contract, contractimpl, Address, Bytes, Env, IntoVal, String};
+use soroban_token_sdk::metadata::TokenMetadata;
+use soroban_token_sdk::TokenUtils;
 
 use crate::error::ContractError;
 use crate::event;
 use crate::storage_types::DataKey;
+use crate::utils::{
+    admin, check_nonnegative_amount, extend_instance_ttl, read_allowance, read_balance,
+    read_decimal, read_name, read_symbol, receive_balance, spend_allowance, spend_balance,
+    write_allowance, write_metadata,
+};
 
 #[contract]
 pub struct InterchainToken;
 
 #[contractimpl]
 impl InterchainToken {
-    pub fn initialize(env: Env, owner: Address) -> Result<(), ContractError> {
+    pub fn initialize_interchain_token(
+        env: Env,
+        interchain_token_service: Address,
+        admin: Address,
+        minter: Address,
+        token_id: Bytes,
+        token_meta_data: TokenMetadata,
+    ) -> Result<(), ContractError> {
+        ensure!(!token_id.is_empty(), ContractError::TokenIdZero);
+        ensure!(token_meta_data.decimal <= 18, ContractError::InvalidDecimal);
+        ensure!(
+            !token_meta_data.name.is_empty(),
+            ContractError::TokenNameEmpty
+        );
```

Comment:
create a `validate_token_metadata` method that does this. it'll be useful elsewhere too